### ISSUE TITLE
More optimal testing strategy

### ISSUE THREAD

**Issue by steven-murray** [21cmfast/21cmFAST#157]
Travis has the ability to separate tests into different "jobs", a feature that we already use a bit (if we ever change to Github Actions, the same is true). We should consider using this to our advantage.

We could consider running some tests less regularly -- eg. only after a PR has been merged to master (this would mean we might get broken code in master, but we would be alerted to it and able to issue a bugfix quickly). Or if we want to be more elaborate and protect master, we could have a "develop" branch which is semi-stable. All of our feature branches are merged into that branch, and it is protected by normal tests, but to merge from develop requires also running the longer regression tests. 

At least, we should probably do a bit of searching on best practices in this area -- we don't want broken code, but we also want our tests to be appropriate at each level.

**Comment by steven-murray** [comment]
@qyx268 also comments that sometimes our tests run so long that they don't fit on Travis. Maybe we should consider buying some time?
